tests/integration/devfile/cmd_devfile_storage_test.go

```diff
@@ -26,373 +26,374 @@ var _ = Describe("odo devfile storage command tests", func() {
+
+				BeforeEach(func() {
+					helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+					stdOut = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+				})
+
+				It("should list storage in pushed state", func() {
+					helper.MatchAllInOutput(stdOut, []string{storageNames[0], pathNames[0], sizes[0], "Pushed"})
+					helper.DontMatchAllInOutput(stdOut, []string{"CONTAINER", "runtime"})
+				})
+
+				When("creating new storage and doing odo storage list", func() {
+					BeforeEach(func() {
+						helper.Cmd("odo", "storage", "create", storageNames[1], "--path", pathNames[1], "--size", sizes[1], "--context", commonVar.Context).ShouldPass()
+						stdOut = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+					})
+
+					It("should list storage in correct state", func() {
+						helper.MatchAllInOutput(stdOut, []string{storageNames[0], pathNames[0], sizes[0], "Pushed"})
+						helper.MatchAllInOutput(stdOut, []string{storageNames[1], pathNames[1], sizes[1], "Not Pushed"})
+						helper.DontMatchAllInOutput(stdOut, []string{"CONTAINER", "runtime"})
+					})
+					When("deleting pushed storage and listing storage", func() {
+						BeforeEach(func() {
+							helper.Cmd("odo", "storage", "delete", storageNames[0], "-f", "--context", commonVar.Context).ShouldPass()
+							stdOut = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+						})
+
+						It("should list it as locally deleted", func() {
+							helper.MatchAllInOutput(stdOut, []string{storageNames[0], pathNames[0], sizes[0], "Locally Deleted"})
+							helper.DontMatchAllInOutput(stdOut, []string{"CONTAINER", "runtime"})
+						})
+
+						When("doing odo push, odo delete -f and listing storage", func() {
+							BeforeEach(func() {
+								helper.Cmd("odo", "push", "--context", commonVar.Context).ShouldPass()
+								helper.Cmd("odo", "delete", "-f", "--context", commonVar.Context).ShouldPass()
+								// since we don't have `wait` for `odo delete` at this moment
+								// we need to wait for the pod to be in the terminating state or it has been deleted from the cluster
+								commonVar.CliRunner.WaitAndCheckForTerminatingState("pods", commonVar.Project, 1)
+								stdOut = helper.Cmd("odo", "storage", "list", "--context", commonVar.Context).ShouldPass().Out()
+							})
+							It("should list storage with correct state", func() {
+								helper.MatchAllInOutput(stdOut, []string{"Not Pushed"})
+								// since `Pushed` is a sub string of `Not Pushed`, we count the occurrence of `Pushed`
+								count := strings.Count(stdOut, "Pushed")
+								Expect(count).To(Equal(1))
+							})
+						})
+					})
+
+					When("a second create the storage and doing odo push", func() {
+						BeforeEach(func() {
+							//helper.Cmd("odo", "storage", "create", storageNames[1], "--size", sizes[1], "--context", commonVar.Context).ShouldPass()
```

Comment:
Why is this commented?

Reply:
Second storage was created in the above step, I have updated `beforeeach()`